core/src/main/java/com/github/phantazmnetwork/core/gui/GuiItem.java

```diff
@@ -0,0 +1,181 @@
+package com.github.phantazmnetwork.core.gui;
+
+import net.minestom.server.Tickable;
+import net.minestom.server.entity.Player;
+import net.minestom.server.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * An interaction-capable item in a GUI. When present in a dynamic GUI, may be animated (its tick method will be called
+ * to update its {@link ItemStack} if necessary, which will then be retrieved through getStack).
+ */
+public interface GuiItem extends Tickable, ClickHandler, RemoveHandler, ReplaceHandler {
+    enum ClickType {
```

Comment:
what's wrong with minestom's

Reply:
some of them are unnecessary for a GUI (like START_DOUBLE_CLICK vs DOUBLE_CLICK, DROP, CHANGE_HELD)
my ClickType is basically a subset of minestom's